### PR TITLE
Developerbox/update2

### PR DESCRIPTION
**Pull request by daniel-thompson**

### CODE REVIEW

File: enterprise/developerbox/build/README.md [hunk 1]
```diff
@@ -1,9 +1,147 @@
...
+
+Run the following commands to clone the source code:
+
+~~~
+cd $WORKSPACE
+git clone git://git.linaro.org/leg/noupstream/arm-trusted-firmware.git -b synquacer
+git clone git://git.linaro.org/leg/noupstream/edk2-platforms.git -b developer-box
+git clone git://git.linaro.org/leg/noupstream/edk2-non-osi.git -b developer-box
+git clone git://git.linaro.org/leg/noupstream/edk2.git -b developer-box
+git clone https://github.com/openssl/openssl  -b OpenSSL_1_1_0e
```
Comment (fboudra) [inline enterprise/developerbox/build/README.md]: Since you already install build dependencies earlier, it isn't necessary to build your own openssl library. fwiw, we install libssl-dev package in the CI. We don't build our own.

File: enterprise/developerbox/build/README.md [hunk 2]
```diff
@@ -1,9 +1,147 @@
...
+a new kernel for your Developerbox we recommend you follow the
+instructions applicable to the distribution you have selected.
+
+Alternatively, advanced users who already have the appropriate compilers
+and headers files installed, may find following generic template useful:
+
+~~~
+git clone https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git
+cd linux
+make defconfig
```
Comment (fboudra) [inline enterprise/developerbox/build/README.md]: aren't we missing ARCH=arm64 here?

File: enterprise/developerbox/build/README.md (hunk 1 above)
Comment (daniel-thompson) [reply enterprise/developerbox/build/README.md]: AFAIK this is source for the crypto routines within EDK2... libssl-dev cannot substitute here.

File: enterprise/developerbox/build/README.md (hunk 2 above)
Comment (daniel-thompson) [reply enterprise/developerbox/build/README.md]: Debatable. The idea behind Developerbox is to develop for Arm on Arm. Thus the default use-case for a kernel build is a self-hosted compile which does not need ARCH to be set explicitly.

Given the aims prefer to assume self-host get as much as we can.

Firmware above is a special case (since we can't self-host if we have burned bad firmware).

File: enterprise/developerbox/build/README.md (hunk 1 above)
Comment (fboudra) [reply enterprise/developerbox/build/README.md]: do you mean our CI produced firmware aren't working?

File: enterprise/developerbox/build/README.md (hunk 2 above)
Comment (fboudra) [reply enterprise/developerbox/build/README.md]: earlier in the document, you say: 

Note: *These instructions will work, without modification, both and x86 PC and on your Developerbox.*

So it isn't consistent. Maybe rewording a bit then?

File: enterprise/developerbox/build/README.md (hunk 1 above)
Comment (daniel-thompson) [reply enterprise/developerbox/build/README.md]: I have made no statements about the CI builds...

What I know (and have just double checked) is that if the git clone of openssl is removed from the above instructions then edk2 for Developerbox fails to compile even with libssl-dev installed.

Are the CI builds you mentioned for Developerbox or for some other platform. If for another platform then does that platform enable CryptoPkg/OpensslLib.

File: enterprise/developerbox/build/README.md (hunk 2 above)
Comment (daniel-thompson) [reply enterprise/developerbox/build/README.md]: That quote is from a different section but, on the grounds that you should not have to tell people how to use documentation, then I will reword this.

**Comment by sdrobertw** [comment]
Can this be merged now?

**Comment by daniel-thompson** [comment]
We should make sure Fathi is happy w.r.t. the openssl dependency.

**Comment by sdrobertw** [comment]
Sounds good, @daniel-thompson - He can merge it when he is ready.

Thanks!
